addons/l10n_ar/models/account_move.py

```diff
@@ -0,0 +1,179 @@
+            if inv.type in ['in_invoice', 'in_refund'] and inv.l10n_latam_document_type_id.purchase_aliquots == 'zero':
+                purchase_aliquots = 'zero'
+            for line in inv.mapped('invoice_line_ids').filtered(lambda x: x.display_type not in ('line_section', 'line_note')):
+                vat_taxes = line.tax_ids.filtered(lambda x: x.tax_group_id.l10n_ar_vat_afip_code)
+                if len(vat_taxes) != 1:
+                    raise UserError(_('There must be one and only one VAT tax per line. Check line "%s"') % line.name)
+                elif purchase_aliquots == 'zero' and vat_taxes.tax_group_id.l10n_ar_vat_afip_code != '0':
+                    raise UserError(_('On invoice id "%s" you must use VAT Not Applicable on every line.')  % inv.id)
+                elif purchase_aliquots == 'not_zero' and vat_taxes.tax_group_id.l10n_ar_vat_afip_code == '0':
+                    raise UserError(_('On invoice id "%s" you must use VAT taxes different than VAT Not Applicable.')  % inv.id)
+
+    # TODO make it with create/write or with https://github.com/odoo/odoo/pull/31059
+    @api.constrains('invoice_date')
+    def set_afip_date(self):
+        for rec in self.filtered('invoice_date'):
+            invoice_date = fields.Datetime.from_string(rec.invoice_date)
+            vals = {}
+            if not rec.l10n_ar_afip_service_start:
+                vals['l10n_ar_afip_service_start'] = invoice_date + relativedelta(day=1)
+            if not rec.l10n_ar_afip_service_end:
+                vals['l10n_ar_afip_service_end'] = invoice_date + relativedelta(day=1, days=-1, months=+1)
+            if vals:
+                rec.write(vals)
+
+    @api.onchange('partner_id')
+    def check_afip_responsibility(self):
+        if self.company_id.country_id == self.env.ref('base.ar') and self.l10n_latam_use_documents and self.partner_id \
+           and not self.partner_id.l10n_ar_afip_responsibility_type_id:
+            return {'warning': {
+                'title': 'Missing Partner Configuration',
+                'message': 'Please configure the AFIP Responsibility for "%s" in order to continue' % (
+                    self.partner_id.name)}}
+
+    def get_document_type_sequence(self):
+        """ Return the match sequences for the given journal and invoice """
+        self.ensure_one()
+        if self.journal_id.l10n_latam_use_documents and self.l10n_latam_country_code == 'AR':
+            if self.journal_id.l10n_ar_share_sequences:
+                return self.journal_id.l10n_ar_sequence_ids.filtered(
+                    lambda x: x.l10n_ar_letter == self.l10n_latam_document_type_id.l10n_ar_letter)
+            res = self.journal_id.l10n_ar_sequence_ids.filtered(
+                lambda x: x.l10n_latam_document_type_id == self.l10n_latam_document_type_id)
+            return res
+        return super().get_document_type_sequence()
+
+    # TODO make it with create/write or with https://github.com/odoo/odoo/pull/31059
```

Comment:
Still TODO...  A constraint is not an onchange and an inverse neither

Reply:
Hi  @jco-odoo 
The requested changes has been applied

Best Regards